cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -180,313 +152,99 @@ handleSimpleNode p NodeCLIArguments{..}
-            -- This means we don't utilise full duplex connection.
-            (Just (Socket.SockAddrInet 0 0))
-            -- IPv6 address
-            (Just (Socket.SockAddrInet6 0 0 (0, 0, 0, 1) 0))
-            (const Nothing)
-            DnsSubscriptionTarget {
-                dstDomain  = BSC.pack raAddress
+          Nothing -> error $ "handleSimpleNode: own address "
+                          <> show myNodeAddress
+                          <> " not found in topology"
+
+    traceWith tracer $ unlines
+      [ "**************************************"
+      , "I am Node "        <> show myNodeAddress
+      , "My producers are " <> show producers'
+      , "**************************************"
+      ]
+
+    let ipProducerAddrs  :: [NodeAddress]
+        dnsProducerAddrs :: [RemoteAddress]
+        (ipProducerAddrs, dnsProducerAddrs) = partitionEithers
+          [ maybe (Right ra) Left $ remoteAddressToNodeAddress ra
+          | ra <- producers' ]
+
+        ipProducers :: [SockAddr]
+        ipProducers = nodeAddressToSockAddr <$> ipProducerAddrs
+
+        dnsProducers :: [DnsSubscriptionTarget]
+        dnsProducers =
+          [ DnsSubscriptionTarget
+              { dstDomain  = BSC.pack raAddress
               , dstPort    = raPort
               , dstValency = raValency
               }
-            (\sock ->
-              connectToNode'
-                (\(DictVersion codec) -> encodeTerm codec)
-                (\(DictVersion codec) -> decodeTerm codec)
-                nullTracer
-                Peer
-                (initiatorNetworkApplication <$> networkAppNodeToNode)
-                sock)
-            wait
-
-      void $ Async.waitAny (localServer : peerServer : ipSubscriptions : dnsSubscriptions)
+          | RemoteAddress {raAddress, raPort, raValency} <- dnsProducerAddrs ]
+
+        myLocalSockPath :: FilePath
+        myLocalSockPath = localSocketFilePath myNodeId
+
+        myLocalAddr :: AddrInfo
+        myLocalAddr = localSocketAddrInfo myLocalSockPath
+
+        runNetworkArgs :: RunNetworkArgs Peer blk
+        runNetworkArgs = RunNetworkArgs
+          { rnaIpSubscriptionTracer  = showTracing $ ipSubscriptionTracer  nodeTracers
+          , rnaDnsSubscriptionTracer = showTracing $ dnsSubscriptionTracer nodeTracers
+          , rnaDnsResolverTracer     = showTracing $ dnsResolverTracer     nodeTracers
+          , rnaMkPeer                = Peer
+          , rnaMyAddr                = nodeAddressInfo myNodeAddress
+          , rnaMyLocalAddr           = myLocalAddr
+          , rnaIpProducers           = ipProducers
+          , rnaDnsProducers          = dnsProducers
+          }
 
+    removeStaleLocalSocket myLocalSockPath
+
+    dbPath <- canonicalizePath =<< makeAbsolute ccDBPath
+
+    varTip <- atomically $ newTVar GenesisPoint
+
+    Node.run
+      (consensusTracers nodeTracers)
+      -- (withTip varTip $ chainDBTracer nodeTracers)
```

Comment:
Left over comment?

Reply:
Good catch! The line below was tracing using `Show`, which we certainly wouldn't want.